File: sample/use_cocoapods/objectivec/GNAdGoogleRewardMediationAdapterSample/GNAdGoogleRewardMediationAdapterSample/MainViewController.m
//
//  MainViewController.m
//  GNAdGoogleRewardMediationAdapterSample
//

#import "MainViewController.h"
#import "RewardNewViewController.h"
#import "RewardRegacyViewController.h"

@interface MainViewController () <UITextFieldDelegate>

@property (weak, nonatomic) IBOutlet UITextField *adUnitIdView;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    _adUnitIdView.delegate = self;
}

- (IBAction)downButtonRegacyApi:(id)sender {
    NSString* unitId = _adUnitIdView.text;
    if (unitId.length == 0) {
        return;
    }

    RewardRegacyViewController* vc =  [self.storyboard instantiateViewControllerWithIdentifier:@"RewardRegacyViewController"];
    vc.unitId = unitId;
    [self.navigationController pushViewController:vc animated:NO];
}

- (IBAction)downButtonNewApi:(id)sender {
    NSString* unitId = _adUnitIdView.text;
    if (unitId.length == 0) {
        return;
    }

    RewardNewViewController* vc =  [self.storyboard instantiateViewControllerWithIdentifier:@"RewardNewViewController"];
    vc.unitId = unitId;
    [self.navigationController pushViewController:vc animated:NO];
}


#pragma mark: UITextFieldDelegate
-(BOOL) textFieldShouldReturn:(UITextField *)textField {
    [textField resignFirstResponder];
    return YES;
}

@end

File: sample/use_cocoapods/swift/GNAdGoogleRewardAdapterSample/GNAdGoogleRewardAdapterSample/MainViewController.swift
//
//  MainViewController.swift
//  GNAdGoogleRewardAdapterSample
//

import UIKit

class MainViewController: UIViewController {

    @IBOutlet weak var adUnitIdView: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        adUnitIdView.delegate = self
    }

    @IBAction func downButtonRegacyApi(_ sender: Any) {
        let unitId: String! = adUnitIdView.text
        if (unitId.isEmpty) {
            return
        }
        let vc = self.storyboard!.instantiateViewController(withIdentifier: "RewardRegacyViewController") as! RewardRegacyViewController
        vc.unitId = unitId
        self.navigationController?.pushViewController(vc, animated: false)
    }

    @IBAction func downButtonNewApi(_ sender: Any) {
        let unitId: String! = adUnitIdView.text
        if (unitId.isEmpty) {
            return
        }
        let vc = self.storyboard!.instantiateViewController(withIdentifier: "RewardNewViewController") as! RewardNewViewController
        vc.unitId = unitId
        self.navigationController?.pushViewController(vc, animated: false)
    }

}
// MARK: UITextFieldDelegate implementation
extension MainViewController : UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        // Hide the keyboard when press return key in a UITextField
        textField.resignFirstResponder()
        return true
    }

}
